File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        data = {}
        for i in nums:
            data[i] = data.get(i, 0) + 1
        for i in data:
            if data[i]>1:
                return True
        return False

        

File: Data Structures & Algorithms/check-if-array-is-sorted-and-rotated/submission-2.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    check(nums: number[]): boolean {
        let min = 101;
        let start_index = -1
        let unrotate:number[]=[]
        for(let i = 0;i<nums.length;i++){
            if(nums[i]<min){
                min=nums[i];
                start_index = i
            }
        }
        // console.log(min,start_index)
        for(let i =0 ;i<nums.length;i++){
            unrotate.push(nums[(i+start_index)%nums.length])
        }
        console.log(unrotate)
        console.log(nums.sort((a,b)=>a-b))
        return JSON.stringify(unrotate) === JSON.stringify(nums.sort((a,b)=>a-b));
    }
}
